test/test_fakeredis.py

```diff
@@ -25,4529 +21,4692 @@
+def zadd(r, key, d, *args, **kwargs):
+    if REDIS3:
+        return r.zadd(key, d, *args, **kwargs)
+    else:
+        return r.zadd(key, **d)
+
+
+def zincrby(r, key, amount, value):
+    if REDIS3:
+        return r.zincrby(key, amount, value)
+    else:
+        return r.zincrby(key, value, amount)
+
+
+@pytest.fixture(scope="session")
+def is_redis_running():
+    try:
+        r = redis.StrictRedis('localhost', port=6379)
+        r.ping()
+    except redis.ConnectionError:
+        return False
+    else:
+        return True
+
+
+@pytest.fixture(
+    params=[
+        pytest.param('StrictRedis', marks=pytest.mark.real),
+        pytest.param('FakeStrictRedis', marks=pytest.mark.fake)
+    ]
+)
+def create_redis(request):
+    name = request.param
+    if not name.startswith('Fake') and not request.getfixturevalue('is_redis_running'):
+        pytest.skip('Redis is not running')
+    decode_responses = request.node.get_closest_marker('decode_responses') is not None
+
+    def factory(db=0):
+        if name.startswith('Fake'):
+            fake_server = request.getfixturevalue('fake_server')
+            cls = getattr(fakeredis, name)
+            return cls(db=db, decode_responses=decode_responses, server=fake_server)
+        else:
+            cls = getattr(redis, name)
+            return cls('localhost', port=6379, db=db, decode_responses=decode_responses)
+
+    return factory
+
+
+@pytest.fixture
+def r(request, create_redis):
+    r = create_redis(db=0)
+    connected = request.node.get_closest_marker('disconnected') is None
+    if connected:
+        r.flushall()
+    yield r
+    if connected:
+        r.flushall()
+    if hasattr(r, 'close'):
+        r.close()     # Older versions of redis-py don't have this method
+
+
+def test_large_command(r):
+    r.set('foo', 'bar' * 10000)
+    assert r.get('foo') == b'bar' * 10000
+
+
+def test_dbsize(r):
+    assert r.dbsize() == 0
+    r.set('foo', 'bar')
+    r.set('bar', 'foo')
+    assert r.dbsize() == 2
+
+
+def test_flushdb(r):
+    r.set('foo', 'bar')
+    assert r.keys() == [b'foo']
+    assert r.flushdb() is True
+    assert r.keys() == []
+
+
+def test_set_then_get(r):
+    assert r.set('foo', 'bar') is True
+    assert r.get('foo') == b'bar'
+
+
+@redis2_only
+def test_set_None_value(r):
+    assert r.set('foo', None) is True
+    assert r.get('foo') == b'None'
+
+
+def test_set_float_value(r):
+    x = 1.23456789123456789
+    r.set('foo', x)
+    assert float(r.get('foo')) == x
+
+
+def test_saving_non_ascii_chars_as_value(r):
+    assert r.set('foo', 'Ñandu') is True
+    assert r.get('foo') == 'Ñandu'.encode()
+
+
+def test_saving_unicode_type_as_value(r):
+    assert r.set('foo', 'Ñandu') is True
+    assert r.get('foo') == 'Ñandu'.encode()
+
+
+def test_saving_non_ascii_chars_as_key(r):
+    assert r.set('Ñandu', 'foo') is True
+    assert r.get('Ñandu') == b'foo'
+
+
+def test_saving_unicode_type_as_key(r):
+    assert r.set('Ñandu', 'foo') is True
```

Comment:
Should this not be `Ñandú`? 😂

Reply:
No idea, it's code I inherited.

Reply:
This is a Ñandú: https://en.wikipedia.org/wiki/Rhea_(bird)

Just wondered out loud 😄